src/main/GCNode.cpp

```diff
@@ -332,7 +332,18 @@ void GCNode::Marker::operator()(const GCNode* node)
     // Update mark  Beware ~ promotes to unsigned int.
     node->m_rcmms &= static_cast<unsigned char>(~s_mark_mask);
     node->m_rcmms |= s_mark;
-    ++m_marks_applied;
+    node->visitReferents(this);
+}
+
+void GCNode::CountingMarker::operator()(const GCNode* node)
+{
+    if (node->isMarked()) {
+	return;
+    }
+    // Update mark  Beware ~ promotes to unsigned int.
+    node->m_rcmms &= static_cast<unsigned char>(~s_mark_mask);
```

Comment:
@kmillar on second thought I don't think it is a good idea to can call the superclass function because `m_marks_applied` should only be incremented when the node was not marked, and then I'd still have to duplicate the `if` statement.